base_layer/wallet/src/storage/sqlite_db/wallet.rs

```diff
@@ -660,33 +662,37 @@ impl Encryptable<XChaCha20Poly1305> for ClientKeyValueSql {
     }
 
     #[allow(unused_assignments)]
-    fn encrypt(&mut self, cipher: &XChaCha20Poly1305) -> Result<(), String> {
-        self.value = encrypt_bytes_integral_nonce(
+    fn encrypt(self, cipher: &XChaCha20Poly1305) -> Result<Self, String> {
+        let mut client_kv = self.clone();
```

Comment:
same as above, this wont zeroize the encrypted values, it will leave them in memory after this function